Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm

```diff
@@ -1568,6 +1569,7 @@ static void activateSessionCleanup(NetworkSessionCocoa& session, const NetworkSe
 SessionWrapper& NetworkSessionCocoa::sessionWrapperForTask(WebPageProxyIdentifier webPageProxyID, const WebCore::ResourceRequest& request, WebCore::StoredCredentialsPolicy storedCredentialsPolicy, std::optional<NavigatingToAppBoundDomain> isNavigatingToAppBoundDomain)
 {
     auto shouldBeConsideredAppBound = isNavigatingToAppBoundDomain ? *isNavigatingToAppBoundDomain : NavigatingToAppBoundDomain::Yes;
+    // FIXME: The following `isParentProcessAFullWebBrowser` check is inaccurate in Safari on macOS.
```

Comment:
Should we specialise isParentProcessAFullWebBrowser for NetworkProcess&?
Or move the boolean flag from NetworkProcess to AuxiliaryProcess and check it in isParentProcessAFullWebBrowser.

Reply:
Indeed — I was thinking of doing the latter (so that the "web browser" checks in all of the WebKit child processes would be accurate, not just for this case).

However, to limit risk for now, I'll do that as a followup patch in https://webkit.org/b/259056 (which should hopefully fix other similar bugs in macOS Safari).